elements/diff/nuxeo-diff.html

```diff
@@ -436,36 +437,86 @@
           }
         },
 
-        _fetchSchemas: function(document) {
-          var type = document.type;
-          if (typeDataCache[type]) {
-            return Promise.resolve(typeDataCache[type]);
+        /**
+         * Returns a list of all the available schemas and fields.
+         * After the first fetch, the list is cached.
+         */
+        _fetchSchemas: function() {
+          if (schemasCache) {
+            return Promise.resolve(schemasCache);
+          }
+          this.$.schema.path = 'config/schemas';
+          return this.$.schema.get().then(function(response) {
+            schemasCache = response;
+            return schemasCache;
+          });
+        },
+
+        /**
+         * Returns a list of all the schemas and fields for a particular document type.
+         * After the first fetch, the list is cached by document type.
+         * Note: Does NOT take into account dynamic facets.
+         */
+        _fetchSchemasForType: function(type) {
+          if (typesCache[type]) {
+            return Promise.resolve(typesCache[type]);
           }
           this.$.schema.path = 'config/types/' + type;
           return this.$.schema.get().then(function(response) {
-            typeDataCache[response.name] = response; // cache response
+            typesCache[response.name] = response;
             return response;
           });
         },
 
         _fetchCommonSchemas: function(left, right) {
-          // if both have the same type, only do a single fetch
-          if (left && right && left.type === right.type) {
-            return this._fetchSchemas(left).then(function(response) {
-              return response.schemas;
-            });
+          if (!left.schemas || !right.schemas) {
+            // the schemas objects might be undefined when using a (forked) writer pre HF14
```

Comment:
We could add the ticket for future reference
```suggestion
            // XXX: the schemas objects might be undefined when using a (forked) writer pre HF14 (see https://jira.nuxeo.com/browse/NXP-27995)
```